app/models/service.rb

```diff
@@ -154,7 +154,7 @@ def self.options_for_sorted_by
   end
 
   def self.options_for_labels
-    ActsAsTaggableOn::Tag.most_used.map { |t| [t.name, t.name] }.unshift(["All labels", ""])
+    ActsAsTaggableOn::Tag.distinct.sort_by {|tags| -tags.name}.map { |t| [t.name, t.name] }.unshift(["All labels", ""])
```

Comment:
If I sort the results, then map them, is the order guaranteed to be maintained? @em-cd

Reply:
Yes, I think so!

Reply:
Yes, I think so!